src/hapi/pipelines/database/national_risk.py

```diff
@@ -0,0 +1,95 @@
+"""Functions specific to the national risk theme."""
+
+from logging import getLogger
+from typing import Dict
+
+from hapi_schema.db_national_risk import DBNationalRisk
+from hdx.utilities.dictandlist import dict_of_dicts_add
+from sqlalchemy.orm import Session
+
+from . import locations
+from .base_uploader import BaseUploader
+from .metadata import Metadata
+
+logger = getLogger(__name__)
+
+
+class NationalRisk(BaseUploader):
+    def __init__(
+        self,
+        session: Session,
+        metadata: Metadata,
+        locations: locations.Locations,
+        results: Dict,
+    ):
+        super().__init__(session)
+        self._metadata = metadata
+        self._locations = locations
+        self._results = results
+
+    def populate(self):
+        logger.info("Populating national risk table")
+        for dataset in self._results.values():
+            datasetinfo = self._metadata.runner.scrapers[
+                "national_risk_national"
+            ].datasetinfo
+            reference_period_start = datasetinfo["source_date"][
+                "default_date"
+            ]["start"]
+            reference_period_end = datasetinfo["source_date"]["default_date"][
+                "end"
+            ]
+            for admin_level, admin_results in dataset["results"].items():
+                rows = dict()
+                resource_id = admin_results["hapi_resource_metadata"]["hdx_id"]
+                for hxl_tag, values in zip(
+                    admin_results["headers"][1], admin_results["values"]
+                ):
+                    for admin_code, value in values.items():
+                        if hxl_tag == "#risk+class":
+                            value = _get_risk_class_code_from_data(value)
+                        dict_of_dicts_add(rows, admin_code, hxl_tag, value)
+                for location in rows:
```

Comment:
Is the purpose of this additional loop (rather than placing the code in this loop where the `dict_of_dicts_add` line is) to try to ensure that rows are grouped by location rather than by #risk+x when you look at them in the db table?

Reply:
The format of the national risk table is wide rather than long, unlike the other tables ([example](https://docs.google.com/spreadsheets/d/1A3u2ZmMTB0YzIJhdXkvnx3JiOzx7Kjtuq1HNg8e8a_w/edit#gid=37744672)), so I'm essentially transposing the results dictionary so it's location based instead of hxl tag based.  The only other way I could think of to do that was using map to pull out the values of each dictionary for each location, but it was more difficult to understand when I read it back: 
    hxl_tags = admin_results["headers"][1]
    locations = list(admin_results["values"][0].keys())
    for location in locations:
        values = list(map(lambda x: x[location], admin_results["values"]))
        values = {hxl_tag: value for hxl_tag in hxl_tags for value in values}